feature_engine/creation/geo_features.py

```diff
@@ -0,0 +1,437 @@
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.creation import GeoDistanceTransformer
+    >>> X = pd.DataFrame({
+    ...     'origin_lat': [40.7128, 34.0522, 41.8781],
+    ...     'origin_lon': [-74.0060, -118.2437, -87.6298],
+    ...     'dest_lat': [34.0522, 41.8781, 40.7128],
+    ...     'dest_lon': [-118.2437, -87.6298, -74.0060],
+    ... })
+    >>> gdt = GeoDistanceTransformer(
+    ...     lat1='origin_lat', lon1='origin_lon',
+    ...     lat2='dest_lat', lon2='dest_lon',
+    ...     method='haversine', output_unit='km'
+    ... )
+    >>> gdt.fit(X)
+    >>> X = gdt.transform(X)
+    >>> X
+       origin_lat  origin_lon  dest_lat   dest_lon  geo_distance
+    0     40.7128    -74.0060   34.0522  -118.2437   3935.746254
+    1     34.0522   -118.2437   41.8781   -87.6298   2808.517344
+    2     41.8781    -87.6298   40.7128   -74.0060   1144.286561
+    """
+
+    def __init__(
+        self,
+        lat1: str,
+        lon1: str,
+        lat2: str,
+        lon2: str,
+        method: Literal["haversine", "euclidean", "manhattan"] = "haversine",
+        output_unit: Literal["km", "miles", "meters", "feet"] = "km",
+        output_col: str = "geo_distance",
+        drop_original: bool = False,
+    ) -> None:
+
+        # Validate coordinate column names
+        for param_name, param_value in [
+            ("lat1", lat1),
+            ("lon1", lon1),
+            ("lat2", lat2),
+            ("lon2", lon2),
+        ]:
+            if not isinstance(param_value, str):
+                raise ValueError(
+                    f"{param_name} must be a string. Got {type(param_value).__name__}."
+                )
+
+        # Validate method
+        valid_methods = ["haversine", "euclidean", "manhattan"]
+        if method not in valid_methods:
+            raise ValueError(
+                f"method must be one of {valid_methods}. Got '{method}' instead."
+            )
+
+        # Validate output_unit
+        valid_units = ["km", "miles", "meters", "feet"]
+        if output_unit not in valid_units:
+            raise ValueError(
+                f"output_unit must be one of {valid_units}. "
+                f"Got '{output_unit}' instead."
+            )
+
+        # Validate output_col
+        if not isinstance(output_col, str):
+            raise ValueError(
+                f"output_col must be a string. Got {type(output_col).__name__}."
+            )
+
+        _check_param_drop_original(drop_original)
+
+        self.lat1 = lat1
+        self.lon1 = lon1
+        self.lat2 = lat2
+        self.lon2 = lon2
+        self.method = method
+        self.output_unit = output_unit
+        self.output_col = output_col
+        self.drop_original = drop_original
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn parameters.
+
+        Validates that the coordinate columns exist and are numerical.
```

Comment:
If I remember correctly, we don't normally include this sentence in our transformers. The library has grown quite a bit, so I might be wrong. I fail to keep track of everything :)